packages/alfa-rules/src/sia-r55/rule.ts

```diff
@@ -82,15 +82,63 @@ export default Rule.Atomic.of<Page, Group<Element>, Question>({
 export namespace Outcomes {
   export const SameResource = (role: Role.Name) =>
     Ok.of(
-      Diagnostic.of(
-        `No two \`${role}\` have the same name and different content.`
+      WithRole.of(
+        `No two \`${role}\` have the same name and different content.`,
+        role
       )
     );
 
   export const DifferentResources = (role: Role.Name) =>
     Err.of(
-      Diagnostic.of(
-        `Some \`${role}\` have the same name and different content.`
+      WithRole.of(
+        `Some \`${role}\` have the same name and different content.`,
+        role
       )
     );
 }
+
+class WithRole extends Diagnostic {
```

Comment:
Not the most inspired name for a class 🙈 Couldn't find a better idea…

Reply:
I like it! 😄